airbyte-integrations/connectors/source-mixpanel/source_mixpanel/components.py

```diff
@@ -0,0 +1,330 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import time
+from dataclasses import dataclass
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional
+
+import dpath.util
+import requests
+from airbyte_cdk.models import AirbyteMessage, SyncMode, Type
+from airbyte_cdk.sources.declarative.extractors import DpathExtractor
+from airbyte_cdk.sources.declarative.interpolation import InterpolatedString
+from airbyte_cdk.sources.declarative.partition_routers import SubstreamPartitionRouter
+from airbyte_cdk.sources.declarative.requesters import HttpRequester
+from airbyte_cdk.sources.declarative.requesters.paginators.strategies.page_increment import PageIncrement
+from airbyte_cdk.sources.declarative.schema import JsonFileSchemaLoader
+from airbyte_cdk.sources.declarative.schema.json_file_schema_loader import _default_file_path
+from airbyte_cdk.sources.declarative.transformations import RecordTransformation
+from airbyte_cdk.sources.declarative.types import Config, Record, StreamSlice, StreamState
+
+from .source import SourceMixpanel
+from .streams.engage import EngageSchema
+
+
+class MixpanelHttpRequester(HttpRequester):
```

Comment:
Could methods `get_url_base` and `get_request_params` be moved to the `manifest`?

Reply:
bump

Reply:
Is it possible the values aren't passed because the fields are not defined on the custom requester? The fields must be defined as dataclass params to be accessible. Here's the [doc](https://docs.airbyte.com/connector-development/config-based/advanced-topics#custom-components)
> The class must also be a dataclass where each field represents an argument to configure from the yaml file, and an InitVar named parameters.

Reply:
I am not sure what exactly goes wrong but I just add request_parameters/request_headers and they do not work:

```
  requester:
    type: CustomRequester
    class_name: "source_mixpanel.components.MixpanelHttpRequester"
    request_parameters:
        project_id: "{{ config['credentials']['project_id'] }}"
         XXXX: XXXXXX
    request_headers:
        XXXXX: XXXXXX
```